### ISSUE TITLE
How to do regularization?

### ISSUE THREAD

**Issue by Hvass-Labs** [google/prettytensor#45]
I'm not sure how regularization works in PrettyTensor. I've done the following (simplified code):

	x_pretty = pt.wrap(x)

	with pt.defaults_scope(activation_fn=tf.nn.relu, l2loss=0.01):
		y_pred = x_pretty. \
		conv2d(kernel=8, depth=32, stride=4). \
		flatten(). \
		fully_connected(size=512). \
		fully_connected(size=10, activation_fn=None)

	loss_regression = y_pred.l2_regression(target=y_true)

	regularizers = tf.get_collection(tf.GraphKeys.REGULARIZATION_LOSSES)
	loss_regularizer = tf.reduce_sum(regularizers)

	loss = loss_regression + 0.01 * loss_regularizer

	optimizer = tf.train.RMSPropOptimizer(learning_rate=1e-3).minimize(loss)

But I don't think this is the correct way of doing it in PrettyTensor.

Could you modify this code to show how it should be done? And perhaps explain why?

Thanks.

**Comment by Hvass-Labs** [comment]
I've done a bit of experimentation and it actually seems like my above code was correct, with the exception of this line which doubly applied the weight on the regularization loss:

    loss = loss_regression + 0.01 * loss_regularizer

It should just be:

    loss = loss_regression + loss_regularizer

since we already have the weight 0.01 in this line:

    with pt.defaults_scope(activation_fn=tf.nn.relu, l2loss=0.01):

Can you confirm that this code is correct and follows best-practice for using the L2 regularizer with PrettyTensor? Or is there a better way?

Thanks.

**Comment by eiderman** [comment]
That is correct and if you are manually applying the optimizer it is a good
way to do it.

**Comment by Hvass-Labs** [comment]
Thanks.
